larky/src/main/resources/vendor/ISO8583Decoder.star

```diff
@@ -0,0 +1,505 @@
+load("@stdlib//larky", larky="larky")
+load("@stdlib//builtins","builtins")
+load("@stdlib//sets", "sets")
+
+
+def _decode(bytes, packager):
+    result = decode(bytes, packager)
+    # print(result)
+    return result
+
+Decoder = larky.struct(
+    decode=_decode,
+)
+
+
+# DecodedDict = Dict[str, str]
+# EncodedDict = Dict[str, Dict[str, bytes]]
+# SpecDict = Mapping[str, Mapping[str, Any]]
+# def DecodeError(msg,
+#     s,
+#     doc_dec,
+#     doc_enc,
+#     pos,
+#     field,
+# ):
+#     r"""Subclass of ValueError that describes ISO8583 decoding error.
+#
+#     Attributes
+#     ----------
+#     msg : str
+#         The unformatted error message
+#     s : bytes or bytearray
+#         The ISO8583 bytes instance being parsed
+#     doc_dec : dict
+#         Dict containing partially decoded ISO8583 data
+#     doc_enc : dict
+#         Dict containing partially encoded ISO8583 data
+#     pos : int
+#         The start index where ISO8583 bytes data failed parsing
+#     field : str
+#         The ISO8583 field where parsing failed
+#     """
+#
+#     def __init__(
+#         msg,
+#         s,
+#         doc_dec,
+#         doc_enc,
+#         pos,
+#         field,
+#     ):
+#         errmsg = msg+"field "+field+" pos"+ pos
+#         ValueError.__init__(self, errmsg)
+#         self.msg = msg
+#         self.s = s
+#         self.doc_dec = doc_dec
+#         self.doc_enc = doc_enc
+#         self.field = field
+#         self.pos = pos
+#     self = __init__(msg, s, doc_dec, doc_enc, pos, field)
+#
+#     def __reduce__(
+#         ) :
+#         return (
+#             self.__class__,
+#             (self.msg, self.s, self.doc_dec, self.doc_enc, self.pos, self.field),
+#         )
+#     self.__reduce__ = __reduce__
+#     return self
+
+
+def decode(
+    s, spec
+) :
+    r"""Deserialize a bytes or bytearray instance containing
+    ISO8583 data to a Python dict.
+
+    Parameters
+    ----------
+    s : bytes or bytearray
+        Encoded ISO8583 data
+    spec : dict
+        A Python dict defining ISO8583 specification.
+        See :mod:`iso8583.specs` module for examples.
+
+    Returns
+    -------
+    doc_dec : dict
+        Dict containing decoded ISO8583 data
+    doc_enc : dict
+        Dict containing encoded ISO8583 data
+
+    Raises
+    ------
+    DecodeError
+        An error decoding ISO8583 bytearray
+    TypeError
+        `s` must be a bytes or bytearray instance
+
+    Examples
+    --------
+    >>> import pprint
+    >>> import iso8583
+    >>> from iso8583.specs import default_ascii as spec
+    >>> s = b"02004010100000000000161234567890123456123456111"
+    >>> doc_dec, doc_enc = iso8583.decode(s, spec)
+    >>> pprint.pprint(doc_dec)
+    {'12': '123456',
+     '2': '1234567890123456',
+     '20': '111',
+     'p': '4010100000000000',
+     't': '0200'}
+    """
+
+    if type(s) not in ("bytes", "bytearray"):
```

Comment:
You can use https://github.com/verygoodsecurity/starlarky/blob/master/larky/src/main/resources/stdlib/base64.star#L13 and https://github.com/verygoodsecurity/starlarky/blob/master/larky/src/main/resources/stdlib/base64.star#L39